### DISCUSSION TITLE
Application.FunctionalTests Fail

### DISCUSSION THREAD

**Discussion by apolonas** [jasontaylordev/CleanArchitecture#969] [General]
I installed the template successfully. The FunctionalTests are all failing with this error:. Am I missing something?

 Message: 
   OneTimeSetUp: System.ArgumentException : Cannot detect the Docker endpoint. Use either the environment variables or the ~/.testcontainers.properties file to customize your configuration:
   https://dotnet.testcontainers.org/custom_configuration/ (Parameter 'DockerEndpointAuthConfig')

**Comment by jasontaylordev** [comment]
No worries, I know the issue is and can help. I just want to make sure we're on the same page. 
Please can you answer the following questions:

1. What version of the template did you use?
2. What version of the .NET SDK do you have installed?
3. What IDE / Editor are you using & what version?
4. What type of solution did you create - Angular, React or Web API only?

**Reply by apolonas** [reply]
Hello Jason. Here are the answers:

I am using version v7.0.2 of the template. (the one using .NET 8 Preview)
Microsoft.AspNetCore.App 8.0.0-preview.6.23329.11
Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview Version 17.7.0 Preview 6.0
Web API only.
Thank you for your reply. Also if I may suggest, a new training video is probably due because a lot of things have changed in the template, from a few years ago. :)

I have used your version of Clean Architecture for a few years now and it is awesome!

Mario

**Reply by apolonas** [reply]
@jasontaylordev I figured it out. I did not have Docker installed locally. Once I Installed it, it all worked. Thanks!

**Reply by jasontaylordev** [reply]
Glad you got it working. If you prefer you can also test against an existing SQL Server. Take a look at; https://github.com/jasontaylordev/CleanArchitecture/blob/main/tests/Application.FunctionalTests/SqlServerTestDatabase.cs.
This will be a little faster too.
